src/Arcus.Scripting.ActiveDirectory/Scripts/Add-AzADAppRoleAssignment.ps1

```diff
@@ -0,0 +1,71 @@
+param(
+    [Parameter(Mandatory = $true)][string] $ClientId = $(throw "ClientId is required"),
+    [Parameter(Mandatory = $true)][string] $Role = $(throw "Role is required"),
+    [Parameter(Mandatory = $true)][string] $AssignRoleToClientId = $(throw "ClientId to assign the role to is required")
+)
+
+$adApplication = Get-AzADApplication -Filter "AppId eq '$ClientId'"
+if (!$adApplication) { 
+    throw "Active Directory Application for the ClientId '$ClientId' could not be found" 
+}
+$adServicePrincipal = Get-AzADServicePrincipal -Filter "AppId eq '$ClientId'"
+if (!$adServicePrincipal) { 
+    throw "Active Directory Service Principal for the ClientId '$ClientId' could not be found" 
+}
+
+$adApplicationRoleAssignTo = Get-AzADApplication -Filter "AppId eq '$AssignRoleToClientId'"
+if (!$adApplicationRoleAssignTo) { 
+    throw "Active Directory Application for the ClientId '$AssignRoleToClientId' could not be found" 
+}
+$adServicePrincipalRoleAssignTo = Get-AzADServicePrincipal -Filter "AppId eq '$AssignRoleToClientId'"
+if (!$adServicePrincipalRoleAssignTo) { 
+    throw "Active Directory Service Principal for the ClientId '$AssignRoleToClientId' could not be found" 
+}
+
+try {
+    if (-not ($adApplication.AppRole.Value -contains $Role)) {
```

Comment:
Maybe we can use `-notcontains` here to make it a bit more readable?
I see 3 other places that could benefit from this.

Reply:
good one, fixed in [`97ab323` (#336)](https://github.com/arcus-azure/arcus.scripting/pull/336/commits/97ab3237d1a2ce136f48819025f384b4194b7987)